docs/usage.md

```diff
@@ -112,3 +122,202 @@ flags were used during project generation, then those adjustments are made in
+though you can specify those in the
+[`xcodeproj.bazelrc`](#workspace-xcodeprojbazelrc) file.
+
+Below are notes about various commands.
+
+### `build`
+
+To build targets the same way as rules_xcodeproj requires more than just using
+this API, because the `xcodeproj` rule applies a
+[configuration transition](https://bazel.build/extending/config#user-defined-transitions)
+to targets. That means that building targets by specifying their labels will
+build potentially different versions of those targets, and minimally versions
+that have different cache keys.
+
+rules_xcodeproj uses
+[output groups](https://bazel.build/extending/rules#requesting_output_files)
+to "address" these correctly configured targets. It uses a set of private
+output groups, but it also exposes the some [public ones](#output-groups).
+
+To build these output groups with this API you would have to craft an call like
+this (**note:** this is not the recommended way to do this, and might break in
+the future, continue reading after the example for the recommended way):
+
+```
+bazel run //:xcodeproj -- build --remote_download_minimal --output_groups=all_targets //:xcodeproj.generator
+```
+
+This requires knowing the internal name of the generator target (`//:xcodeproj.generator` in this example), and it also doesn't apply some flags that Xcode
+`bazel build` command applies (e.g. `--experimental_remote_download_regex`).
+Instead, it's recommended that you use the
+[`--generator_output_groups` option](#--generator_output_groups):
+
+```
+bazel run //:xcodeproj -- --generator_output_groups=all_targets build --remote_download_minimal
+```
+
+> **Note**
+>
+> You can't have a naked label as part of your command (e.g.
+> `aquery '//some:target'`). This is because bazel will think you are trying to
+> pass that label to the outside `bazel run`. Since you probably want to use
+> [`--generator_output_groups`](#--generator_output_groups) anyway, this
+> shouldn't be a problem. If you need to pass labels though, you can use the
+> [`--target_pattern_file` flag](https://bazel.build/reference/command-line-reference#flag--target_pattern_file).
+
+### `clean`
+
+When you run `bazel clean` normally (i.e. not using this API), it won't affect
+the rules_xcodeproj output base the way you expect. `bazel clean --expunge`
+will though, as it will blow away both environments. To clean the
+rules_xcodeproj output base use the API instead:
+
+```
+bazel run //:xcodeproj -- clean
+```
+
+### `query`/`cquery`/`aquery`
+
+Depending on how you have your [rules_xcodeproj Bazel configs](#bazel-configs)
+set up, you might be able to run `bazel query` without using the API. I
+recommend using the API instead though, to prevent fetching external
+dependencies in the primary output base. The other queries, `cquery` and
+`aquery`, should always be performed through the API, to ensure the targets
+are properly configured:
+
+```
+bazel run //:xcodeproj -- aquery 'set(//some:target)'
```

Comment:
This is nice, but it changes the semantic way of using bazel. I don't think we (me) would want to train people doing the-rules_xcodeproj-way-of-using-bazel.

Reply:
I'm not sure of a way around it. If I don't mention this the error you get is hard to understand, so I have to say _something_.

Reply:
Thinking more about this, I could have the command portion quoted, similar to when one does `sh -c`. Then it would be something like `bazel run //:xcodeproj -- 'query //some:target'` which bazel wouldn't mess with.